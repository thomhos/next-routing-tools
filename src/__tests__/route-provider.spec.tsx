import React from 'react'
import propTypes from 'prop-types'
import { create } from 'react-test-renderer'

import { route, RoutesMap, RouteProvider } from '../'

class ProviderTester extends React.Component {
    public context: { routes: RoutesMap }

    static contextTypes = {
        routes: propTypes.object,
    }

    public render() {
        return <div>routes: {JSON.stringify(this.context.routes)}</div>
    }
}

describe('RouteProvider', () => {
    it('Has a context type', () => {
        expect(RouteProvider.childContextTypes).toHaveProperty('routes')
    })

    it('Renders', () => {
        const routes = {
            index: route('/', '/'),
        }
        const comp = create(
            <RouteProvider routes={routes}>
                <ProviderTester />
            </RouteProvider>,
        )
        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
