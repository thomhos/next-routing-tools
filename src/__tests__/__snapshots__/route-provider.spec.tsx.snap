// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteProvider Renders 1`] = `
<div>
  routes: 
  {"index":{"_keys":[],"_path":"/","_regex":{},"_page":"/"}}
</div>
`;
